File: voiceover/src/server/ollama_response.rs
use serde::Deserialize;
use serde::Serialize;
#[derive(Serialize, Deserialize)]
pub(crate) struct OllamaResponse {
    pub model: String,
    pub created_at: String,
    pub response: String,
    pub done: bool,
    pub context: Vec<i32>,
    pub total_duration: i64,
    pub load_duration: i32,
    pub prompt_eval_duration: i64,
    pub eval_count: i32,
    pub eval_duration: i64,
}

File: voiceover/model/ollama_response.go
package model

type Ollama_response struct {
	Model                string  `json:"model"`
	Created_at           string  `json:"created_at"`
	Response             string  `json:"response"`
	Done                 bool    `json:"done"`
	Context              []int32 `json:"context"`
	Total_duration       int64   `json:"total_duration"`
	Load_duration        int32   `json:"load_duration"`
	Prompt_eval_duration int64   `json:"prompt_eval_duration"`
	Eval_count           int32   `json:"eval_count"`
	Eval_duration        int64   `json:"eval_duration"`
}
